tensorboard/webapp/runs/views/runs_selector/runs_selector_container.ts

```diff
@@ -49,6 +71,44 @@ export class RunsSelectorContainer {
       ].filter((col) => col !== null) as RunsTableColumn[];
     })
   );
+  private readonly ngUnsubscribe = new Subject();
 
   constructor(private readonly store: Store<State>) {}
+
+  ngOnInit() {
+    // Notify the user that new runs may not be selected. Avoid showing it too
+    // often, since it would be annoying to see the alert re-appear on every
+    // auto-reload (assuming a new run per reload).
+    const runsExceedsLimitForRoute$ = this.experimentIds$.pipe(
+      takeUntil(this.ngUnsubscribe),
+      switchMap(() => {
+        // Returns an Observable that emits once and completes when the current
+        // route's run count goes over the limit.
+        return this.store.select(getCurrentRouteRunSelection).pipe(
+          distinctUntilChanged((a, b) => {
+            return a === b;
+          }),
+          filter((runSelectionMap: Map<string, boolean> | null) => {
+            if (!runSelectionMap) {
+              return false;
+            }
+            return runSelectionMap.size > MAX_NUM_RUNS_TO_ENABLE_BY_DEFAULT;
+          }),
+          take(1)
+        );
+      })
+    );
+    runsExceedsLimitForRoute$.subscribe(() => {
+      const text =
+        `The number of runs exceeds ` +
+        `${MAX_NUM_RUNS_TO_ENABLE_BY_DEFAULT}. New runs are unselected ` +
+        `for performance reasons.`;
+      this.store.dispatch(alertReported({localizedMessage: text}));
+    });
```

Comment:
this is effectively creating a composite action; please do this in the alerts feature.

Reply:
As per offline discussion, moved the composite action to RunsTable, and added a comment to indicate that this should be considered an exception, which goes against recommended patterns.